vra-services-check/src/testsuite_module/vrahealthtestsuite.py

```diff
@@ -0,0 +1,236 @@
+
+        try:
+             response = requests.request("POST", url, data=payload1, headers=headers, verify=False)
+             responseJSON = response.json()
+             if responseJSON != None:
+                 if  responseJSON.get('id'):
+                     token = responseJSON['id']
+                     #print("DEBUG: extracted vRA login token:!")
+                 else:
+                     #print ("DEBUG: failed to authenticate to vRA system tenant, API URL: "+url+ " as user: "+user+"\n")
+                     token = None 
+             else:
+                 #print ("DEBUG: no HTTP response received from vRA system tenant, API URL: "+url)
+                 token = None
+
+        except Exception as exc:
+             #print("Error: vRA login to: "+url+" failed with error message:" + str(exc))
+             raise Exception("Failed to login to vRA system tenant  URL: "+url+ ", tenant: "+tenant+" as user: "+user+"\n")
+        return token
+
+    #end of login to vRA tenant function definition 
+
+
+    #start function definition of 'get_registered_services'
+    def get_registered_services(self,url,user,pwd,tenant):
+   
+        querystring = {"page":"1","limit":"20"}
+        payload = "providing the currently requested pagination and sorting parameters"
+        headers = {'Accept': "application/json",'Content-Type': "application/json",'Cache-Control': "no-cache"}
+        # first obtain a valid token by calling login_vra_token
+        theToken = self.login_vra_token(url+"/identity/api/tokens",user,pwd,tenant)
+        
+        if theToken != None:
+            headers = {'Accept': "application/json",'Authorization': "Bearer " + theToken,'Cache-Control': "no-cache"}
+            response = requests.request("GET", url+"/component-registry/services", data=payload, headers=headers, params=querystring, verify=False)
+            if response:
+
+               #print("DEBUG OK - Obtained vRA Registered Services response \n")
+               servicesJSON = json.loads(response.content.decode('utf-8'))
+               #print("DEBUG OK - parsed services into JSON format \n")
+               return servicesJSON['content']
+            else:
+               #print("DEBUG, PROBLEM: failed to obtain vRA Registered Services!")
+               #return response
+               raise Exception("Failed to obtain vRA registered Services for tenant: " + tenant)
+        else:
+            #print("DEBUG, PROBLEM: vRA login failed - cannot retrieve services data!")
+            return None
+
+    #end of get_registered_services function definition
+
+    #start Test suite method 'check_registered_services' definition 
+    @test_name("Check number of vRA Registered services")
+    @test_description("Checks that core vRA Services are Registered Post Upgrade")
+    @test_access_level(AccessLevel.NORMAL.value)
+    @test_severity(Severity.NORMAL.value)
+    @test_remediation("https://www.vmware.com/products/vrealize-automation.html")
```

Comment:
Consider a more specific remediation link.  Perhaps to the services section of the documentation?

Reply:
@test_remediation("https://www.vmware.com/products/vrealize-automation.html") used for each of test cases to point to specific area of vRA 7.x documentation? Should I use 7.5 docs?

Reply:
yes, you'll have to use the current version (e.g. 7.5)